File: backend/recover.js
const fs = require('fs');
const logs = fs.readFileSync('/home/gaurishankar/.gemini/antigravity/brain/edac7fd4-4fa5-49c9-aa6c-573870c974cb/.system_generated/logs/overview.txt', 'utf8').split('\n').filter(l => l.trim().length > 0);

for (const line of logs) {
  try {
    const json = JSON.parse(line);
    if (json.tool_calls) {
      for (const call of json.tool_calls) {
        if (call.name === 'run_command' && call.args && call.args.CommandLine) {
          const cmd = call.args.CommandLine;
          if (cmd.includes("cat << 'EOF' > backend/patternEngine.js")) {
            const content = cmd.split("cat << 'EOF' > backend/patternEngine.js\\n")[1].split("\\nEOF")[0];
            fs.writeFileSync('patternEngine_recovered.js', content.replace(/\\n/g, '\n').replace(/\\\\"/g, '"'));
          }
          if (cmd.includes("cat << 'EOF' > backend/backtestEngine.js")) {
            const content = cmd.split("cat << 'EOF' > backend/backtestEngine.js\\n")[1].split("\\nEOF")[0];
            fs.writeFileSync('backtestEngine_recovered.js', content.replace(/\\n/g, '\n').replace(/\\\\"/g, '"'));
          }
          if (cmd.includes("cat << 'EOF' > backend/slTpCalculator.js")) {
             const content = cmd.split("cat << 'EOF' > backend/slTpCalculator.js\\n")[1].split("\\nEOF")[0];
             fs.writeFileSync('slTpCalculator_recovered.js', content.replace(/\\n/g, '\n').replace(/\\\\"/g, '"'));
          }
        }
      }
    }
  } catch (e) {}
}
console.log("Recovery attempted.");

File: backend/recover.py
import json

with open('/home/gaurishankar/.gemini/antigravity/brain/edac7fd4-4fa5-49c9-aa6c-573870c974cb/.system_generated/logs/overview.txt', 'r') as f:
    for line in f:
        line = line.strip()
        if not line:
            continue
        try:
            data = json.loads(line)
            if 'tool_calls' in data:
                for call in data['tool_calls']:
                    if call.get('name') == 'run_command':
                        cmd = call.get('args', {}).get('CommandLine', '')
                        if "cat << 'EOF' > backend/patternEngine.js" in cmd:
                            content = cmd.split("cat << 'EOF' > backend/patternEngine.js\\n")[1].split("\\nEOF")[0]
                            content = content.replace('\\n', '\n').replace('\\"', '"')
                            with open('patternEngine_recovered.js', 'w') as out: out.write(content)
                        if "cat << 'EOF' > backend/backtestEngine.js" in cmd:
                            content = cmd.split("cat << 'EOF' > backend/backtestEngine.js\\n")[1].split("\\nEOF")[0]
                            content = content.replace('\\n', '\n').replace('\\"', '"')
                            with open('backtestEngine_recovered.js', 'w') as out: out.write(content)
                        if "cat << 'EOF' > backend/slTpCalculator.js" in cmd:
                            content = cmd.split("cat << 'EOF' > backend/slTpCalculator.js\\n")[1].split("\\nEOF")[0]
                            content = content.replace('\\n', '\n').replace('\\"', '"')
                            with open('slTpCalculator_recovered.js', 'w') as out: out.write(content)
                        if "cat << 'EOF' > backend/monitorWorker.js" in cmd:
                            content = cmd.split("cat << 'EOF' > backend/monitorWorker.js\\n")[1].split("\\nEOF")[0]
                            content = content.replace('\\n', '\n').replace('\\"', '"')
                            with open('monitorWorker_recovered.js', 'w') as out: out.write(content)
        except Exception as e:
            pass

print("Recovery attempted.")
